Fails to locate qscintilla2.prf on Fedora 21

Hi,

Newly released OpenSCAD 2015.03 fails to see /usr/share/qt5/mkspecs/features/qscintilla2.prf and thus fails to build due to the "lack" of QScintilla2 installed.

I have the following packages in my system:

qscintilla-qt5-2.8.4-2.fc21.i686
qscintilla-qt5-devel-2.8.4-2.fc21.i686 

And the qscintilla2.prf file is exactly where I said it is.

What do I need to tweak to get this fixed?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9426311-fails-to-locate-qscintilla2-prf-on-fedora-21?utm_campaign=plugin&utm_content=tracker%2F52063&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F52063&utm_medium=issues&utm_source=github).
</bountysource-plugin>

t-paul's  link  command did indeed work.   Now both openscad's  check-dependencies.sh  command and  qmake openscad.pro  are happy.

However, OpenSCAD still doesn't build because my version of boost is newer than expected?   check-dependencies says that boost 1.35 is needed, and correctly detects that I have boost 1.41.  But there is an error:

g++ -c -pipe -fno-strict-aliasing -frounding-math -DEIGEN_DONT_ALIGN -I/usr/local/include/glib-2.0 -I/usr/local/lib64/glib-2.0/include -I/usr/local/include/harfbuzz -I/usr/local/include/freetype2 -I/usr/local/include -I/usr/local/include -I/usr/local/include/freetype2 -I/usr/local/include/libxml2 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -O2 -D_REENTRANT -Wall -W -Wno-unused-local-typedefs -fPIE -DOPENSCAD_VERSION=2015.03-1 -DOPENSCAD_SHORTVERSION=2015.03 -DOPENSCAD_YEAR=.0 -DOPENSCAD_MONTH=.0 -DOPENSCAD_DAY=.0 -DDEBUG -DENABLE_CGAL -DENABLE_OPENCSG -DUSE_SCINTILLA_EDITOR -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CONCURRENT_LIB -DQT_CORE_LIB -I/usr/lib64/qt5/mkspecs/linux-g++ -I. -Isrc -isystem /usr/include/qt5 -Isrc/libtess2/Include -isystem /usr/include/qt5/QtOpenGL -isystem /usr/include/qt5/QtPrintSupport -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtConcurrent -isystem /usr/include/qt5/QtCore -Iobjects -isystem /usr/include/eigen3 -o objects/scintillaeditor.o src/scintillaeditor.cpp
In file included from src/boosty.h:30:0,
                 from src/scintillaeditor.cpp:5:
src/scintillaeditor.cpp: In constructor EditorColorScheme::EditorColorScheme(boost::filesystem::path):
src/scintillaeditor.cpp:79:59: error: boost::filesystem::path has no member named c_str
   PRINTB("Error reading color scheme file '%s': %s", path.c_str() % e.what());

That should be   path.string().c_str()

Ditto in all of the calls to the PRINTDB macro in src/PlatformUtils.cc


Likewise, 

src/scintillaeditor.cpp: In member function void ScintillaEditor::enumerateColorSchemesInPath(ScintillaEditor::colorscheme_set_t&, boost::filesystem::path):
src/scintillaeditor.cpp:373:27: error: boost::filesystem::basic_path<std::basic_string<char>, boost::filesystem::path_traits>::string_type has no member named string
    if (!(path.extension().string() == ".json")) {

should probably be:

    if (!(path.extension() == ".json")) {

Ditto in src/colormap.cc  for both  extension()  and  PRINTB  as shown above.


With these four edits, OpenSCAD appears to compile and run correctly with boost 1.41 (Nov 17 2009).  

The latest boost is 1.58.0 (Apr 17 2015); but I have 1.41 since that is what ships with RHEL 6.6,  2.6.32-504.3.3.el6.x86_64 (my company is very conservative in RHEL updates).

Thanks,

Scott

--- a/qscintilla2.prf
+++ b/qscintilla2.prf
@@ -10,6 +10,16 @@ INCLUDEPATH += $$[QT_INSTALL_HEADERS]
 
 LIBS += -L$$[QT_INSTALL_LIBS]
 
+QT5LIB=qt5scintilla2
+
+unix:linux* {
+    FEDORA32LIBS=/usr/lib/libqscintilla2-qt5.so
+    FEDORA64LIBS=/usr/lib64/libqscintilla2-qt5.so
+    exists($$FEDORA32LIBS) | exists($$FEDORA64LIBS) {
+        QT5LIB=qscintilla2-qt5
+    }
+}
+
 CONFIG(debug, debug|release) {
     mac: {
 	#LIBS += -lqscintilla2_debug
@@ -19,7 +29,8 @@ CONFIG(debug, debug|release) {
 	    LIBS += -lqscintilla2d
 	} else {
 	    greaterThan(QT_MAJOR_VERSION, 4) {
-		LIBS += -lqt5scintilla2
+		message("Using $$QT5LIB as library name")
+		LIBS += -l$$QT5LIB
 	    } else {
 		LIBS += -lqscintilla2
 	    }
@@ -33,7 +44,8 @@ CONFIG(debug, debug|release) {
         LIBS += -lqscintilla2
       } else {
         greaterThan(QT_MAJOR_VERSION, 4) {
-	    LIBS += -lqt5scintilla2
+	    message("Using $$QT5LIB as library name")
+	    LIBS += -l$$QT5LIB
         } else {
 	    LIBS += -lqscintilla2
         }

